### PR TITLE
Fix recurrence time equality check for singular-plural form

### LINKED ISSUE #306: Fix duplicate tasks not detected if recurrence time is in singular/plural form
Currently, a task with recurrence `3 day` and `3 days` are considered different.

image

### LINKED ISSUE #350: [PE-D][Tester A] No error message shown when invalid recurring date is parsed
When adding a task using the command:

`add -p 1 d/ new task with bad recurring date| 28-10-22 | abcd`

where "abcd" is an invalid recurring date format, no error message is shown. 

-------------
Labels: `severity.Low` `type.FunctionalityBug`
original: edricyeo/ped#4

### PR DESCRIPTION
**Pull request by rama-pang**
Closes #306. Additionally closes #350.

### CODE REVIEW

**Comment by izzahaj** [comment]
Looking at issue #306, perhaps a different fix may be in order, since frequency should always be plural if frequency != 1. e.g. the app should never save the frequency string as `3 day`. Even if user inputs `3 day` (because it's accepted), it should always be converted to plural form `3 days`.

The other fix (for #350) looks good.

File: src/main/java/seedu/uninurse/model/task/RecurringTask.java [hunk 1]
```diff
@@ -114,6 +120,46 @@ public boolean equals(Object other) {
...
                 && this.getDateTime().equals(o.getDateTime())
-                && this.recurrence.equals(o.recurrence);
+                && this.isSimilarRecurrence(recurrence, o.recurrence);
+    }
+
+    private boolean isSimilarRecurrence(Recurrence recurrence1, Recurrence recurrence2) {
+        if (recurrence1.toString().equals(recurrence2.toString())) {
+            return true; // if both are equal
+        }
+        if ((recurrence1.toString() + "S").equals(recurrence2.toString())) {
```
Comment (izzahaj) [inline src/main/java/seedu/uninurse/model/task/RecurringTask.java:130]: Given the new changes (e.g. `2 day` input taken as `2 days` internally, would this check be redundant?

File: src/main/java/seedu/uninurse/model/task/RecurringTask.java [hunk 2]
```diff
@@ -114,6 +120,46 @@ public boolean equals(Object other) {
...
+            return true; // if recurrence1 is singular, recurrence2 is plural
+        }
+        if ((recurrence2.toString() + "S").equals(recurrence1.toString())) {
```
Comment (izzahaj) [inline src/main/java/seedu/uninurse/model/task/RecurringTask.java:133]: same as above

File: src/main/java/seedu/uninurse/model/task/RecurringTask.java [hunk 3]
```diff
@@ -114,6 +120,46 @@ public boolean equals(Object other) {
...
+            }
+        }
+        return false;
+    }
+
+    private static Recurrence getSingularRecurrence(Recurrence recurrence) {
+        if (isSingularRecurrence(recurrence)) {
+            return recurrence;
+        }
+        String string = recurrence.toString();
```
Comment (BlopApple) [inline src/main/java/seedu/uninurse/model/task/RecurringTask.java:152]: Maybe declaring a string is redundant?

File: src/main/java/seedu/uninurse/model/task/RecurringTask.java [hunk 4]
```diff
@@ -114,6 +120,46 @@ public boolean equals(Object other) {
...
+        System.out.println(string.substring(0, string.length() - 1));
+        return Recurrence.valueOf(string.substring(0, string.length() - 1));
+    }
+
+    private static Recurrence getPluralRecurrence(Recurrence recurrence) {
+        if (!isSingularRecurrence(recurrence)) {
+            return recurrence;
+        }
+        String string = recurrence.toString();
```
Comment (BlopApple) [inline src/main/java/seedu/uninurse/model/task/RecurringTask.java:161]: Same as above.

**Review by BlopApple — CHANGES_REQUESTED** [review]

File: src/main/java/seedu/uninurse/model/task/RecurringTask.java (hunk 1 above)
Comment (BlopApple) [reply src/main/java/seedu/uninurse/model/task/RecurringTask.java:130]: 'Izzah pointed out a good point and I also think that this check is possibly redundant.

**Comment by rama-pang** [comment]
Additionally fixed `RecurringTask` not checking for `frequency` when comparing equality.

**Review by izzahaj — APPROVED** [review]
LGTM

**Review by BlopApple — APPROVED** [review]
LGTM

**Pull request merged**